tests/emukit/quadrature/test_square_transform_bq.py

```diff
@@ -0,0 +1,276 @@
+    return emukit_method
+
+def square_transform_bq_incorrect_bound():
+    X = np.array([[-2, 0], [0, 1], [3, 2]])
+    Y = np.array([[4], [0], [9]])
+    dim = X.shape[1]
+    bound = 1  # intentionally incorrect
+
+    gpy_model = GPy.models.GPRegression(X=X, Y=Y, kernel=GPy.kern.RBF(input_dim=dim))
+    measure = IsotropicGaussianMeasure(mean=np.ones(dim), variance=1.)
+    qrbf = QuadratureRBFIsoGaussMeasure(RBFGPy(gpy_model.kern), measure=measure)
+    model = BaseGaussianProcessGPy(kern=qrbf, gpy_model=gpy_model)
+
+    emukit_method = LinSquareTransformRBFIsoGaussMeasureBayesianQuadrature(base_gp=model,
+                                                                                X=X, Y=Y,
+                                                                                bound=bound)
+
+def test_square_transform_bq_shapes(square_transform_bq):
+    X = np.array([[5, 0], [-2, 1]])
+    Y = np.array([[-25.1], [-4.1]])
+
+    # integrate
+    res = square_transform_bq.integrate()
+    assert len(res) == 2
+    assert isinstance(res[0], float)
+    assert isinstance(res[1], float)
+
+    # transformations
+    assert square_transform_bq.transform(Y).shape == Y.shape
+    assert square_transform_bq.inverse_transform(Y).shape == Y.shape
+
+    # predictions base
+    res = square_transform_bq.predict_base(X)
+    assert len(res) == 4
+    for i in range(4):
+        assert res[i].shape == (X.shape[0], 1)
+
+    # predictions base full covariance
+    res = square_transform_bq.predict_base_with_full_covariance(X)
+    assert len(res) == 4
+    assert res[0].shape == (X.shape[0], 1)
+    assert res[1].shape == (X.shape[0], X.shape[0])
+    assert res[2].shape == (X.shape[0], 1)
+    assert res[3].shape == (X.shape[0], X.shape[0])
+
+    # predictions
+    res = square_transform_bq.predict(X)
+    assert len(res) == 2
+    assert res[0].shape == (X.shape[0], 1)
+    assert res[1].shape == (X.shape[0], 1)
+
+    # predictions full covariance
+    res = square_transform_bq.predict_with_full_covariance(X)
+    assert len(res) == 2
+    assert res[0].shape == (X.shape[0], 1)
+    assert res[1].shape == (X.shape[0], X.shape[0])
+
+    # predict gradients
+    res = square_transform_bq.get_prediction_gradients(X)
+    assert len(res) == 2
+    assert res[0].shape == (X.shape[0], X.shape[1])
+    assert res[1].shape == (X.shape[0], X.shape[1])
+
+
+def test_square_transform_bq_transformations():
+    X = np.random.rand(5, 2)
+    Y = np.array([[1], [2], [3], [4], [5]])
+    bound = 6
+
+    mock_kern = mock.create_autospec(QuadratureRBFIsoGaussMeasure)
+    mock_gp = mock.create_autospec(IBaseGaussianProcess, kern=mock_kern, X=X, Y=Y)
+
+    # no alpha
+    method = LinSquareTransformRBFIsoGaussMeasureBayesianQuadrature(base_gp=mock_gp, X=X, Y=Y, bound=bound)
+
+    assert_allclose(method.inverse_transform(Y), np.array([[3.16227766], [2.82842712], [2.44948974],
+                                                            [2.], [1.41421356]]))
```

Comment:
could we use some more obvious numbers? Otherwise it is not clear of the test is correct. e.g., numbers which are integers both transformed and not transformed.